sdk/core/azure-core/inc/azure/core/http/websockets/websockets.hpp

```diff
@@ -0,0 +1,369 @@
+       */
+      WebSocketTextFrame(bool isFinalFrame, uint8_t const* body, size_t size)
+          : WebSocketFrame{WebSocketFrameType::TextFrameReceived, isFinalFrame},
+            Text(body, body + size)
+      {
+      }
+    };
+
+    /** @brief Contains the contents of a WebSocket Binary frame.*/
+    class WebSocketBinaryFrame : public WebSocketFrame,
+                                 public std::enable_shared_from_this<WebSocketBinaryFrame> {
+      friend Azure::Core::Http::WebSockets::_detail::WebSocketImplementation;
+
+    private:
+    public:
+      /** @brief Constructs a new WebSocketBinaryFrame */
+      WebSocketBinaryFrame() : WebSocketFrame(WebSocketFrameType::BinaryFrameReceived){};
+      /** @brief Binary frame data received from the remote peer. */
+      std::vector<uint8_t> Data;
+
+      /** @brief Constructs a new WebSocketBinaryFrame
+       * @param isFinal True if this is the final frame in a multi-frame message.
+       * @param body binary of the frame data.
+       * @param size Length in bytes of the frame body.
+       */
+    private:
+      WebSocketBinaryFrame(bool isFinal, uint8_t const* body, size_t size)
+          : WebSocketFrame{WebSocketFrameType::BinaryFrameReceived, isFinal},
+            Data(body, body + size)
+      {
+      }
+    };
+
+    /** @brief Contains the contents of a WebSocket Close frame.*/
+    class WebSocketPeerCloseFrame : public WebSocketFrame,
+                                    public std::enable_shared_from_this<WebSocketPeerCloseFrame> {
+      friend Azure::Core::Http::WebSockets::_detail::WebSocketImplementation;
+
+    public:
+      /** @brief Constructs a new WebSocketPeerCloseFrame */
+      WebSocketPeerCloseFrame() : WebSocketFrame(WebSocketFrameType::PeerClosedReceived){};
+      /** @brief Status code sent from the remote peer. Typically a member of the WebSocketErrorCode
+       * enumeration */
+      uint16_t RemoteStatusCode{};
+      /** @brief Optional text sent from the remote peer. */
+      std::string RemoteCloseReason;
+
+    private:
+      /** @brief Constructs a new WebSocketBinaryFrame
+       * @param remoteStatusCode Status code sent by the remote peer.
+       * @param remoteCloseReason Optional reason sent by the remote peer.
+       */
+      WebSocketPeerCloseFrame(uint16_t remoteStatusCode, std::string const& remoteCloseReason)
+          : WebSocketFrame{WebSocketFrameType::PeerClosedReceived},
+            RemoteStatusCode(remoteStatusCode), RemoteCloseReason(remoteCloseReason)
+      {
+      }
+    };
+
+    struct WebSocketOptions : Azure::Core::_internal::ClientOptions
+    {
+      /**
+       * @brief The set of protocols which are supported by this client
+       */
+      std::vector<std::string> Protocols = {};
+
+      /**
+       * @brief The protocol name of the service client. Used for the User-Agent header
+       * in the initial WebSocket handshake.
+       */
+      std::string ServiceName;
+      /**
```

Comment:
either add new line, or remove the other new lines between elements, it's inconsistent